### PR TITLE
Removing exception logging on HexBlock.verifyBlockDims

### LINKED ISSUE #2557: Remove log message about confirming block dimensions
These log messages
```
 Block design <[BLOCK]> is too complicated to verify dimensions. Make sure they are correct!
```
flood the logs. 

The cuplit is the aptly named `verifyBlockDims` method

https://github.com/terrapower/armi/blob/6e4ac3925a04ec73c64de20166a95ec78aeb9be6/armi/reactor/blocks/hexBlock.py#L332-L341

which works if and only if

1. you have one and only one wire component
2. you have one and only one clad component
3. you have one or more duct components

Otherwise, the log is produced.

I see the appeal of an automated dimension checker. But I would contend this is too strict to be on the "general purpose" `HexBlock`.

Applications bringing their own `Block` subclasses could specialize this implementation for a given design.

### PR DESCRIPTION
**Pull request by yardasol**
## What is the change? Why is it being made?

Removes logging from try-except block in `HexBlock.verifyBlockDims` due to excessive logging described in #2557, and replaces it with a silent exit. 
Explain in the function docstring the cases in which `HexBlock.verifyBlockDims` will silent exit.

closes #2557 

## SCR Information

Change Type: trivial

One-Sentence Rationale: Better explains the behavior of `HexBlock.verifyBlockDims` and removes excessive logging.

One-line Impact on Requirements: NA

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Comment by yardasol** [comment]
I tried to look for a unit test that checked `HexBlock.verifyBlockDims`, but couldn't find it. I checked the coverage of the change via
 
```bash
coverage run --rcfile=pyproject.toml -m pytest -n 4 --cov=armi/reactor --cov-config=pyproject.toml --cov-report=xml --ignore=venv armi\reactor\tests\test_blocks.py
```
and it seems that the affected lines are still covered in `coverage.xml`:
```xml
<?xml version="1.0" ?>
<coverage version="7.13.4" timestamp="1782404434639" lines-valid="9759" lines-covered="5668" line-rate="0.5808" branches-covered="0" branches-valid="0" branch-rate="0" complexity="0">
        <!-- Generated by coverage.py: https://coverage.readthedocs.io/en/7.13.4 -->
        <!-- Based on https://raw.githubusercontent.com/cobertura/web/master/htdocs/xml/coverage-04.dtd -->
        <sources>
                <source>C:\Users\olyardas\codes\armi\armi\reactor</source>
        </sources>
        <packages>
                <package name="." line-rate="0.5664" branch-rate="0" complexity="0">
                        <classes>
                                ...
                                <class name="hexBlock.py" filename="blocks/hexBlock.py" complexity="0" line-rate="0.8033" branch-rate="0">
                                        <methods/>
                                        <lines>
                                                ...
                                                <line number="339" hits="1"/>
                                                <line number="340" hits="1"/>
                                                <line number="341" hits="1"/>
                                                <line number="342" hits="1"/>
                                                <line number="343" hits="1"/>
                                                <line number="347" hits="1"/>
```
Please let me know if I need to do more.

**Comment by john-science** [comment]
> I tried to look for a unit test that checked `HexBlock.verifyBlockDims`...

You just removed existing code. There's no reason to worry about code coverage. 

The fun part is that sometimes a PR like this will drop the code coverage by a line or two. And there's just not much for it.

**Comment by yardasol** [comment]
> You just removed existing code. There's no reason to worry about code coverage.

I was a bit worried because it doesn't look like there's a test that covers `HexBlock.verifyBlockDims` in the first place, but I guess it's still covered by the suite (based on the coverage).

File: armi/reactor/blocks/hexBlock.py [hunk 1]
```diff
@@ -330,14 +330,20 @@ def _rotateDisplacement(self, rad: float):
...
         try:
             wireComp = self.getComponent(Flags.WIRE, quiet=True)  # Quiet because None case is checked for below
             ductComps = self.getComponents(Flags.DUCT)
             cladComp = self.getComponent(Flags.CLAD, quiet=True)  # Quiet because None case is checked for below
         except ValueError:
-            # there are probably more that one clad/wire, so we really dont know what this block looks like
-            runLog.info(f"Block design {self} is too complicated to verify dimensions. Make sure they are correct!")
+            # silent exit
+            # there is no duct or more that one clad/wire, the logic below cannot
+            # check the dimensions of the block.
```
Comment (john-science) [inline armi/reactor/blocks/hexBlock.py]:
```suggestion
            # Silent Exit
            # There is no duct or more that one clad/wire, the logic below cannot check the dimensions of the block.
```

Comment (john-science) [reply armi/reactor/blocks/hexBlock.py]: ARMI's `ruff` sets the max line length to 120 characters. We might as well use it. #shrug #not-PR-blocking

**Review by john-science — APPROVED** [review]

**Comment by john-science** [comment]
> I was a bit worried because it doesn't look like there's a test that covers `HexBlock.verifyBlockDims` in the first place

Ahhhh

Well, no, that's no good.  If the method is staying, we should directly cover it with a test or two. Make sure we hit some edge cases.

> Would you rather have that be part of this PR or a separate one?

**Comment by yardasol** [comment]
@john-science Thatshould be part of this PR IMO. I'll try to whip something up

**Comment by yardasol** [comment]
@john-science Okay, after a long time exploring the test file, there is a bigger issue at play than I thought. Inconsistencies between the unit tests classes + missing unit tests. I've done a more "proper" writeup in [this issue](https://github.com/terrapower/armi/issues/2565), but I really think this should be in a different PR.
